server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -142,7 +151,7 @@ protected void parseCreateField(ParseContext context, Consumer<IndexableField> o
             return;
         }
         if (fieldType().isSearchable() || fieldType.stored()) {
-            onField.accept(new Field(fieldType().name(), value ? "T" : "F", fieldType));
+            onField.accept(new Field(fieldType().name(), value ? "T" : "F", fieldTypeWithoutDocValues));
```

Comment:
Since now the default doc-value type is `SORTED_NUMERIC`, `Field` with non-numeric value causes exceptions.